test/functional/sqlsrv/MsHelper.inc

```diff
@@ -370,32 +425,106 @@ function insertRow($conn, $tbname, $inputs, &$r = null, $api = INSERT_QUERY)
         $params = array();
         foreach ($inputs as $key => $input) {
             if (is_object($input)) {
-                array_push($params, $input->bindParamArr($inputs[$key]));               
+                array_push($params, $input->bindParamArr($input->value));
             } else {
                 array_push($params, $inputs[$key]);
             }
         }
-        
+
         // use prepare for inserts when AE is enabled
         if (isColEncrypted() || $api == INSERT_PREPARE_PARAMS) {
             $stmt = sqlsrv_prepare($conn, $insertSql, $params);
-            $r = sqlsrv_execute($stmt);
+            if ($stmt) {
+                $r = sqlsrv_execute($stmt);
+            }
         } else {
             $stmt = sqlsrv_query($conn, $insertSql, $params);
-        } 
+        }
+    }
+    return $stmt;
+}
+
+/**
+ * Perform a simple select from a table with or without where condition(s)
+ * @param  resource  $conn : connection resource
+ * @param  string  $tbname : name of the table
+ * @param  string  $conds : string of condition(s) with placeholders, null by default
+ * @param  array  $values : array of parameters, null by default
+ * @return object sqlsrv statement upon success or false otherwise
+ */
+function selectFromTable($conn, $tbname, $conds = null, $values = null)
+{
+    $sql = "SELECT * FROM $tbname";
+    return executeQuery($conn, $sql, $conds, $values);
+}
+
+/**
+ * Perform a query from a table with or without where condition(s)
+ * @param  resource  $conn : connection resource
+ * @param  string  $sql : T-SQL query
+ * @param  string  $conds : string of condition(s) with placeholders, null by default
+ * @param  array  $values : array of parameters, null by default
+ * @return object sqlsrv statement upon success or false otherwise
+ */
+function executeQuery($conn, $sql, $conds = null, $values = null)
+{
+    // when AE is enabled, use sqlsrv_prepare() to handle fields with unlimited size
```

Comment:
how can you tell if a field has unlimited size from here? If AE is enabled, does it always use sqlsrv_prepare?

Reply:
yes, I'll modify the comments accordingly